src/main/java/org/maxgamer/quickshop/Listeners/PlayerListener.java

```diff
@@ -348,22 +341,22 @@ public void onTeleport(PlayerTeleportEvent e) {
     @EventHandler(priority = EventPriority.HIGH, ignoreCancelled = true)
     public void onDyeing(PlayerInteractEvent e) {
         if (e.getAction() != Action.RIGHT_CLICK_BLOCK ||
-            e.getItem() == null ||
-            !Util.isDyes(e.getItem().getType())) {
+                e.getItem() == null ||
+                !Util.isDyes(e.getItem().getType())) {
             return;
         }
 
         final Block block = e.getClickedBlock();
 
         if (block == null ||
-            !Util.isWallSign(block.getType())) {
+                !Util.isWallSign(block.getType())) {
             return;
         }
 
         final Block attachedBlock = Util.getAttached(block);
 
         if (attachedBlock == null ||
-            plugin.getShopManager().getShopIncludeAttached(attachedBlock.getLocation()) == null) {
+                plugin.getShopManager().getShopIncludeAttached(attachedBlock.getLocation()) == null) {
```

Comment:
Don't check attached by code self, get shop include attached already did it.
Double check will harm the performance.
@portlek